.github/workflows/CI.yml

```diff
@@ -0,0 +1,120 @@
+  workflow_dispatch:
+
+jobs:
+  # JOB to run change in the build files
+  changes:
+    runs-on: ubuntu-latest
+    # Required permissions
+    permissions:
+      pull-requests: read
+    # Set job outputs to values from filter step
+    outputs:
+      files: ${{ steps.filter.outputs.files }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Filter files
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 #v3.0.2
+        id: filter
+        with:
+          filters: |
+            files:
+              - 'pyproject.toml'
+              - 'setup.py'
+              - '.conda/env_build.yml'
+              - '.conda/meta.yml'
+
+  verify-conda-build:
+    name: Conda Build
+    runs-on: ubuntu-latest
+    needs: changes
+    # Only run if there are changes in the build files
+    if: ${{ needs.changes.outputs.files == 'true' }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup conda build environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: "3.10" # TODO: Add back in vars.PY_VERSION
+          environment-file: .conda/env_build.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          show-channel-urls: true
+
+      - name: Verify conda recipe
+        shell: bash -el {0}
+        run: conda-verify .conda --ignore C2105,C2122
+
+      - name: Run conda build
+        shell: bash -el {0}
+        run: |
+          conda build . --no-anaconda-upload --output-folder=./build -c conda-forge -c accessnri -c coecms
+
+      - name: Verify conda package
+        shell: bash -el {0}
+        run: conda-verify ./build/noarch/*.tar.bz2 --ignore C1105,C1115,C1141
+  
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Setup conda environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ matrix.python-version }}
+          environment-file: .conda/env_dev.yml
+          activate-environment: yamanifest-dev
+          auto-update-conda: false
+          auto-activate-base: false
+          show-channel-urls: true
+
+      - name: Install source
+        shell: bash -l {0}
+        run: python3 -m pip install --no-deps --no-build-isolation -e .
+
+      - name: List installed packages
+        shell: bash -l {0}
+        run: conda list
+
+      # - name: Lint # TODO: Add back in linting 
+      #   shell: bash -l {0}
+      #   run: pylint --extension-pkg-whitelist=netCDF4 --ignored-modules=yamanifest -E yamanifest
```

Comment:
Does this need to be added back in this PR?

Reply:
I’ve just removed the pylint dependency as it was what was removed recently in `um2nc-standalone` repository which I used as a reference for the conda deployment CI/CD (https://github.com/ACCESS-NRI/um2nc-standalone/issues/167). It would be good to set up a pre-commit hook for linting at some point (if organisational wide linting CI workflows are not set up), but I think that should be a separate PR